src/Vault.sol

```diff
@@ -128,22 +128,21 @@ contract Vault is IVault, VaultToken, Ownable {
             paid = msg.value;
         }
 
-        // subtraction must be safe
-        SettlementGuard.accountDelta(msg.sender, currency, -(paid.toInt128()));
+        SettlementGuard.accountDelta(msg.sender, currency, paid.toInt128());
     }
 
     /// @inheritdoc IVault
     function settleFor(Currency currency, address target, uint256 amount) external isLocked {
         /// @notice settle all outstanding debt if amount is 0
-        /// It will revert if target has negative delta
-        if (amount == 0) amount = SettlementGuard.getCurrencyDelta(target, currency).toUint256();
-        SettlementGuard.accountDelta(msg.sender, currency, amount.toInt128());
-        SettlementGuard.accountDelta(target, currency, -(amount.toInt128()));
+        /// It will revert if target has positive delta
```

Comment:
curious, what does this comment mean now. does it mean that it will revert if the target has non-zero delta? 

eg. the comment might imply that if target has positive delta, this method will revert

Reply:
when calling `settleFor`, `msg.sender` is trying to cover the payment for target right ? 


If the target has a positive delta, it means that the target doesn't owe anything to the pool and, in fact, the pool owes tokens to them..(after updating everything flip) In this case, pass in amount=0 (i.e. ettle all outstanding debt for target) is invalid.

Reply:
> when calling settleFor, msg.sender is trying to cover the payment for target right ?

I guess when there's a comment like ` /// It will revert if target has positive delta` -- people might think there's a revert in this method. But its basically the `CurrencyNotSettled()` revert that happens at the end right? or is there actually a validation for this `target has positive delta` specifically?

Reply:
“But its basically the CurrencyNotSettled() revert that happens at the end right? or is there actually a validation for this target has positive delta specifically?”

No, it will revert as in `SafeCast.toUint256` because we cant cast a negative number (given delta is positive, minus it results in negative) to uint256

![CleanShot 2024-05-28 at 15 58 06@2x](https://github.com/pancakeswap/pancake-v4-core/assets/137024020/2e7f88b5-a09a-4c21-9731-7abaef8dc93a)